packages/autocomplete/src/containers/AutocompleteContainer.js

```diff
@@ -0,0 +1,540 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import scrollTo from 'dom-helpers/util/scrollTo';
+import styled from 'styled-components';
+import { Manager, Target, Popper } from 'react-popper';
+import { Portal } from 'react-portal';
+import Fragment from 'render-fragment';
+import {
+  ControlledComponent,
+  composeEventHandlers,
+  SingleSelectionModel,
+  IdManager,
+  KEY_CODES
+} from '@zendeskgarden/react-selection';
+import { requiredParam } from '@zendeskgarden/react-utilities';
+import { getPopperPlacement, getRtlPopperPlacement } from '@zendeskgarden/react-tooltips';
+import { withTheme, isRtl } from '@zendeskgarden/react-theming';
+
+/**
+ * This container must provide a wrapper for the provided menu
+ * due to constraints in our arrow css. We must ensure that the container
+ * of the menu can retain its relative positioning. Without this
+ * container Popper would apply absolute positioning.
+ */
+const MenuWrapper = styled.div`
+  z-index: ${props => props.zIndex};
+`;
+
+class AutocompleteContainer extends ControlledComponent {
+  static propTypes = {
+    /**
+     * Appends the menu to the provided element
+     */
+    appendToNode: PropTypes.instanceOf(Element),
```

Comment:
For this to be used in SSR it would have to be [pony-filled](https://ponyfoo.com/articles/polyfills-or-ponyfills) (my new vocab for the day) anyway.

Would it make more sense to just use `PropTypes.node` rather than a custom validator (since it wouldn't match in SSR anyway)?